sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -694,37 +698,94 @@ private[cosmos] class CosmosRowConverter(
   }
 
     private def parseTtlExpired(objectNode: ObjectNode): Boolean = {
-      getFullFidelityMetadata(objectNode) match {
-        case metadataNode: Some[ObjectNode] =>
-          metadataNode.get.get(TimeToLiveExpiredPropertyName) match {
-            case valueNode: JsonNode =>
-              Option(valueNode).fold(false)(v => v.asBoolean(false))
-            case _ => false
-          }
+      getAttributeNode(objectNode, MetadataJsonBodyAttributeName) match {
+        case metadataNode: JsonNode =>
+          metadataNode.get(TimeToLiveExpiredPropertyName) match {
+        case valueNode: JsonNode =>
+          Option(valueNode).fold(false)(v => v.asBoolean(false))
         case _ => false
+        }
+    case _ => false
       }
     }
 
-    private def parseOperationType(objectNode: ObjectNode): String = {
-      getFullFidelityMetadata(objectNode) match {
-        case metadataNode: Some[ObjectNode] =>
-          metadataNode.get.get(OperationTypePropertyName) match {
+    private def parseId(objectNode: ObjectNode): String = {
+        var currentNode = getAttributeNode(objectNode, CurrentAttributeName)
+        if (currentNode == null || currentNode.isEmpty) {
+            currentNode = getAttributeNode(objectNode, PreviousRawJsonBodyAttributeName)
+        }
```

Comment:
factor out finding the "right" payload node into separate function to avoid code duplication

Reply:
ack, will do.